streaming/rrdpush.c

```diff
@@ -1174,10 +1174,10 @@ static int rrdpush_receive(int fd
 
     info("STREAM %s [receive from [%s]:%s]: initializing communication...", host->hostname, client_ip, client_port);
     char initial_response[HTTP_HEADER_SIZE];
-    if (stream_flags & STREAM_HAS_INTEGER_VERSION) {
+    if (stream_version > 2) {
         info("STREAM %s [receive from [%s]:%s]: Netdata is using the newest stream protocol.", host->hostname, client_ip, client_port);
-        sprintf(initial_response, "%s%u", START_STREAMING_PROMPT_VN, STREAMING_PROTOCOL_CURRENT_VERSION);
-    } else if (stream_flags & LABEL_FLAG_UPDATE_STREAM) {
+        sprintf(initial_response, "%s%u", START_STREAMING_PROMPT_VN, stream_version);
+    } else if (stream_version > 0) {
```

Comment:
Are `old`, `second` and `newest`  good names? Why do not use version numbers.

And `stream_version > 0` condition looks suspicious, i guess it can be only `1`. Am i right?

Where could i read about stream versioning? Is there a document or an discussion in some issue at least?

Reply:
> Are `old`, `second` and `newest` good names? Why do not use version numbers.

I removed the words `second` and `newest` and I am using version now.

> And `stream_version > 0` condition looks suspicious, i guess it can be only `1`. Am i right?

We are in the third version already, unfortunately due the previous versions do not have the version explicit, we "lost" the count. You were right, I also had a mistake, but I fixed with another commit.

> Where could i read about stream versioning? Is there a document or a discussion in some issue at least?

This was not documented in the past, but you can see more details about where we are in this link https://github.com/netdata/netdata/issues/7804#issue-552508664 .

Reply:
Ok, i see, so it is `0-3`.